diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/workgroup.pm mezzanine_patched_e-smith-samba-1.14.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/workgroup.pm
--- e-smith-samba-1.14.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/workgroup.pm	2003-04-10 13:14:32.000000000 -0600
+++ mezzanine_patched_e-smith-samba-1.14.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/workgroup.pm	2007-01-10 07:49:22.000000000 -0700
@@ -154,7 +154,7 @@
     my $workgroup = lc(shift);
     
     my $server = lc($fm->cgi->param('ServerName'));
-    return "INVALID_WORKGROUP" unless ( $workgroup =~ /^([a-zA-Z][\-\w\.]*)$/ );
+    return "INVALID_WORKGROUP" unless ( $workgroup =~ /^([a-zA-Z0-9][\-\w\.]*)$/ );
     return 'INVALID_WORKGROUP_MATCHES_SERVERNAME' if ( $server eq $workgroup);
     return ('OK'); 
     
