diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/40vetoOplock mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/40vetoOplock
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/40vetoOplock	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/40vetoOplock	2007-01-25 14:58:47.000000000 -0700
@@ -0,0 +1,9 @@
+{
+    if ( ($ibay->prop('OpLocks') || 'enabled') eq 'disabled') {
+	$OUT .= "oplocks = no\n";
+	$OUT .= "level2 oplocks = no";
+    } else {
+	$vetofiles = $ibay->prop('VetoOplockFiles') || return '';
+	$OUT = "veto oplock files = $vetofiles";
+    }
+}
