diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/createlinks mezzanine_patched_e-smith-samba-1.14.0/createlinks
--- e-smith-samba-1.14.0/createlinks	2007-10-11 21:11:03.000000000 -0400
+++ mezzanine_patched_e-smith-samba-1.14.0/createlinks	2007-10-11 21:10:11.000000000 -0400
@@ -89,6 +89,9 @@
 $event = "machine-account-create";
 event_link("create-machine-account", $event, "10");
 
+$event = "pre-restore";
+event_link("delete-smbpasswd", $event, "90");
+
 # Link smbd and nmbd to daemontools.
 safe_symlink("daemontools", "root/etc/rc.d/init.d/smbd");
 safe_symlink("daemontools", "root/etc/rc.d/init.d/nmbd");
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete-smbpasswd mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete-smbpasswd
--- e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete-smbpasswd	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete-smbpasswd	2007-10-11 21:09:13.000000000 -0400
@@ -0,0 +1,2 @@
+#! /bin/sh
+exec rm -f /etc/samba/smbpasswd
