diff -up e-smith-samba-2.2.0/root/home/e-smith/files/server-resources/regedit/win7samba.reg.win7samba e-smith-samba-2.2.0/root/home/e-smith/files/server-resources/regedit/win7samba.reg
--- e-smith-samba-2.2.0/root/home/e-smith/files/server-resources/regedit/win7samba.reg.win7samba	2009-12-26 09:47:18.000000000 +0100
+++ e-smith-samba-2.2.0/root/home/e-smith/files/server-resources/regedit/win7samba.reg	2009-12-26 09:47:55.000000000 +0100
@@ -0,0 +1,8 @@
+Windows Registry Editor Version 5.00
+
+[HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\LanManWorkstation\Parameters]
+"DNSNameResolutionRequired"=dword:00000000
+"DomainCompatibilityMode"=dword:00000001
+
+[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\services\Netlogon\Parameters
+"RequireStrongKey"=dword:00000000
