diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/createlinks mezzanine_patched_e-smith-samba-1.14.0/createlinks
--- e-smith-samba-1.14.0/createlinks	2005-06-16 00:31:24.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.14.0/createlinks	2006-04-10 18:53:00.158538588 +1000
@@ -80,10 +80,10 @@
 safe_symlink("sigterm", "root/etc/e-smith/events/$event/services2adjust/nmbd");
 
 $event = "printer-create";
-safe_symlink("sighup", "root/etc/e-smith/events/$event/services2adjust/smbd");
+safe_symlink("sigterm", "root/etc/e-smith/events/$event/services2adjust/smbd");
 
 $event = "printer-delete";
-safe_symlink("sighup", "root/etc/e-smith/events/$event/services2adjust/smbd");
+safe_symlink("sigterm", "root/etc/e-smith/events/$event/services2adjust/smbd");
 
 $event = "machine-account-create";
 event_link("create-machine-account", $event, "10");
