diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/events/actions/update-domain-group-maps mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/update-domain-group-maps
--- e-smith-samba-1.14.0/root/etc/e-smith/events/actions/update-domain-group-maps	2007-01-30 09:58:07.000000000 -0500
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/update-domain-group-maps	2007-01-30 09:57:46.000000000 -0500
@@ -12,13 +12,13 @@
 
 my $a = esmith::AccountsDB->open_ro or die "Couldn't open accounts db\n";
 
-my $g = `net getlocalsid`;
+my $g = `/usr/bin/net getlocalsid`;
 unless ($g =~ /SID.*is: (.+)/) {
     warn "Unable to determine SID.  Clearning cache to see if it helps.";
     rename '/etc/samba/secrets.tdb','/etc/samba/secrets.'.time;
     rename '/var/cache/samba/gencache.tdb','/var/cache/samba/gencache.'.time;
     rename '/var/cache/samba/wins.dat','/var/cache/samba/wins.'.time;
-    $g = `net getlocalsid`;
+    $g = `/usr/bin/net getlocalsid`;
     $g =~ /SID.*is: (.+)/ or die "Could not get current sid\n";
 }
 my $local_sid = $1;
