diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/var/service/nmbd/run mezzanine_patched_e-smith-samba-1.14.0/root/var/service/nmbd/run
--- e-smith-samba-1.14.0/root/var/service/nmbd/run	2005-03-11 09:41:55.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/var/service/nmbd/run	2007-02-16 13:28:59.000000000 -0700
@@ -12,7 +12,7 @@
 if [ "$status" = "disabled" ]
 then
     /sbin/e-smith/config setprop nmbd status disabled
-    runsvctrl d .
+    sv d .
     exit
 fi
 
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/var/service/smbd/run mezzanine_patched_e-smith-samba-1.14.0/root/var/service/smbd/run
--- e-smith-samba-1.14.0/root/var/service/smbd/run	2007-02-16 13:49:52.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/var/service/smbd/run	2007-02-16 13:29:02.000000000 -0700
@@ -12,7 +12,7 @@
 if [ "$status" = "disabled" ]
 then
     /sbin/e-smith/config setprop smbd status disabled
-    runsvctrl d .
+    sv d .
     exit
 fi
 
