diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-2.2.0/root/etc/e-smith/templates/etc/smb.conf/11lanmanPasswords mezzanine_patched_e-smith-samba-2.2.0/root/etc/e-smith/templates/etc/smb.conf/11lanmanPasswords
--- e-smith-samba-2.2.0/root/etc/e-smith/templates/etc/smb.conf/11lanmanPasswords	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-samba-2.2.0/root/etc/e-smith/templates/etc/smb.conf/11lanmanPasswords	2010-09-23 18:12:48.000000000 +0200
@@ -0,0 +1,5 @@
+{
+
+$OUT .= "lanman auth = $smb{'LanManPasswords'}\n" if $smb{'LanManPasswords'};
+
+}
