diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/defaults/smb/ServerName mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/defaults/smb/ServerName
--- e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/defaults/smb/ServerName	2005-02-17 15:11:34.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/defaults/smb/ServerName	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-mitel-networks-server
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/migrate/30smbServerName mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/migrate/30smbServerName
--- e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/migrate/30smbServerName	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/db/configuration/migrate/30smbServerName	2007-01-26 18:07:19.000000000 -0700
@@ -0,0 +1,8 @@
+{
+    my $smb = $DB->get('smb') || return;
+    my $srvName = $smb->prop('ServerName') || 'e-smith-server';
+    my $sysName = $DB->get('SystemName') || return;
+
+    return unless $srvName =~ m#(mitel-networks|e-smith)-server#;
+    $smb->set_prop('ServerName', $sysName->value);
+}
