--- e-smith-samba-2.2.0/root/etc/e-smith/events/actions/create-machine-account.fixe_create_machine_accounts	2010-11-10 19:46:19.000000000 +0100
+++ e-smith-samba-2.2.0/root/etc/e-smith/events/actions/create-machine-account	2010-11-10 19:46:55.000000000 +0100
@@ -63,7 +63,6 @@
 warn "create-machine-account $machineName: Creating Unix user and group\n";
 
 if ($ldapauth ne 'enabled')
-else
 {
     system(
             "/usr/sbin/useradd",
