diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/createlinks mezzanine_patched_e-smith-samba-1.14.0/createlinks
--- e-smith-samba-1.14.0/createlinks	2007-10-10 15:27:22.000000000 -0400
+++ mezzanine_patched_e-smith-samba-1.14.0/createlinks	2007-10-10 15:26:44.000000000 -0400
@@ -83,6 +83,7 @@
 safe_symlink("sigterm", "root/etc/e-smith/events/$event/services2adjust/smbd");
 
 $event = "printer-delete";
+event_link("delete_printer_tdb", $event, 15);
 safe_symlink("sigterm", "root/etc/e-smith/events/$event/services2adjust/smbd");
 
 $event = "machine-account-create";
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete_printer_tdb mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete_printer_tdb
--- e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete_printer_tdb	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/events/actions/delete_printer_tdb	2007-10-10 15:24:57.000000000 -0400
@@ -0,0 +1,6 @@
+#! /bin/sh
+
+ACTION=$1
+PRINTER=$2
+
+exec rm -f /var/cache/samba/printing/$PRINTER.tdb
