diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/ibays/40browseable mezzanine_patched_e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/ibays/40browseable
--- e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/ibays/40browseable	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/ibays/40browseable	2008-01-09 11:26:30.000000000 +1100
@@ -0,0 +1,5 @@
+{
+    if ( ($ibay->prop('Browseable') || 'yes') eq 'disabled') {
+        $OUT .= "browseable = no\n";
+    }
+}
