diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare	2005-02-17 16:04:02.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare	2007-01-25 13:49:59.000000000 -0700
@@ -12,5 +12,6 @@
 browseable = no
 create mask = 0600
 directory mask = 0700
+csc policy = disable
 HERE
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/35cscPolicy mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/35cscPolicy
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/35cscPolicy	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/35cscPolicy	2007-01-25 13:48:43.000000000 -0700
@@ -0,0 +1,4 @@
+{
+    $policy = $ibay->prop('cscPolicy') || return '';
+    $OUT = "csc policy = $policy";
+}
