diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare mezzanine_patched_e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare
--- e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare	2007-12-15 08:53:08.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.1/root/etc/e-smith/templates/etc/smb.conf/61Profilesshare	2008-03-21 11:06:20.000000000 -0600
@@ -13,5 +13,6 @@
 create mask = 0600
 directory mask = 0700
 csc policy = disable
+hide files = /desktop.ini/ntuser.ini/NTUSER.*/
 HERE
 }
