diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11unixPasswordSync mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11unixPasswordSync
--- e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11unixPasswordSync	2002-08-01 12:09:24.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11unixPasswordSync	2006-01-25 16:36:58.184597716 +1100
@@ -8,3 +8,6 @@
 }
 unix password sync = Yes
 pam password change = Yes
+
+passwd program = /usr/bin/passwd %u 
+passwd chat = *password* %nn *password* %nn *successfully*
