diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.2/root/etc/e-smith/db/configuration/defaults/smb/OpLocks mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/db/configuration/defaults/smb/OpLocks
--- e-smith-samba-1.13.2/root/etc/e-smith/db/configuration/defaults/smb/OpLocks	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/db/configuration/defaults/smb/OpLocks	2005-12-14 14:57:40.492592053 +1100
@@ -0,0 +1 @@
+enabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11kernelOplocks mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11kernelOplocks
--- e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11kernelOplocks	2002-08-08 05:40:51.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11kernelOplocks	1970-01-01 10:00:00.000000000 +1000
@@ -1 +0,0 @@
-kernel oplocks = true
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11level2Oplocks mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11level2Oplocks
--- e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11level2Oplocks	2002-08-08 05:40:51.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11level2Oplocks	1970-01-01 10:00:00.000000000 +1000
@@ -1,4 +0,0 @@
-{
-# The 2.4.x Linux kernel can now synchronise oplock requests
-}
-level2 oplocks = true
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11oplocks mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11oplocks
--- e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11oplocks	2002-08-08 05:40:51.000000000 +1000
+++ mezzanine_patched_e-smith-samba-1.13.2/root/etc/e-smith/templates/etc/smb.conf/11oplocks	2005-12-14 15:02:51.834422073 +1100
@@ -1,4 +1,10 @@
 {
-# The 2.4.x Linux kernel can now synchronise oplock requests
+    my $oplocks = (($smb{OpLocks} || 'enabled') eq 'enabled') 
+			? 'true' : 'false';
+
+    $OUT =<<HERE;
+oplocks = $oplocks
+kernel oplocks = $oplocks
+level2 oplocks = $oplocks
+HERE
 }
-oplocks = true
