diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/var/service/smbd/run mezzanine_patched_e-smith-samba-1.14.0/root/var/service/smbd/run
--- e-smith-samba-1.14.0/root/var/service/smbd/run	2005-03-12 03:41:55.000000000 +1100
+++ mezzanine_patched_e-smith-samba-1.14.0/root/var/service/smbd/run	2006-05-31 14:31:34.905128892 +1000
@@ -33,4 +33,4 @@
 rm -f /var/cache/samba/printing.tdb
 
 exec 2>&1
-exec $smbd -F -S
+exec $smbd -F
