diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir
--- e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir	2005-07-18 22:39:57.595330344 -0600
+++ mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir	2005-07-18 22:39:30.822400448 -0600
@@ -39,7 +39,7 @@
 foreach my $user ( @newusers )
 {
     die "$user is not a user account\n"
-        unless ( grep /^$user$/ @users );
+        unless ( grep /^$user$/, @users );
 
     my $dir = "/home/e-smith/files/samba/profiles/$user";
    
