diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/00usedb mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/00usedb
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/00usedb	2002-03-12 15:19:39.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/00usedb	2007-01-06 14:40:34.000000000 -0700
@@ -1,3 +1,5 @@
 {
     use esmith::db;
+
+    $vfs = ();
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10recyclebin mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10recyclebin
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10recyclebin	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10recyclebin	2007-01-06 14:38:19.000000000 -0700
@@ -0,0 +1,11 @@
+{
+    $OUT = "";
+    return unless (($smb{'RecycleBin'} || 'disabled') eq 'enabled');
+
+    $vfs->{recycle}->{versions} = ($smb{'KeepVersions'} || 'disabled') eq 'enabled' ?  "True" : "False";
+    $vfs->{recycle}->{repository} = "Recycle Bin";
+    $vfs->{recycle}->{keeptree} = "True";
+    $vfs->{recycle}->{touch} = "True";
+    $vfs->{recycle}->{exclude} = "*.tmp|*.temp|*.o|*.obj|~\$*";
+    $vfs->{recycle}->{exclude_dir} = "/tmp|/temp|/cache";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10shadowcopy mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10shadowcopy
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10shadowcopy	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/10shadowcopy	2007-01-06 14:39:54.000000000 -0700
@@ -0,0 +1,4 @@
+{
+    $vfs->{shadow_copy} = () if (($smb{'ShadowCopy'} || 'disabled') eq 'enabled');
+    $OUT = "";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesRecycleBin mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesRecycleBin
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesRecycleBin	2006-03-16 00:01:15.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesRecycleBin	1969-12-31 17:00:00.000000000 -0700
@@ -1,18 +0,0 @@
-{
-    $OUT = "";
-    my $bin = $smb{'RecycleBin'} || 'disabled';
-    return unless $bin eq 'enabled';
-
-    my $keep_versions = $smb{'KeepVersions'} || 'disabled';
-    $keep_versions = ($keep_versions eq 'enabled') ? 'True' : 'False';
-
-    $OUT .= "vfs objects = recycle:repository recycle:keeptree " .
-	"recycle:versions recycle:touch recycle:exclude " .
-	"recycle:exclude_dir\n" .
-	"  recycle:repository=Recycle Bin\n" .
-	"  recycle:keeptree=True\n" .
-	"  recycle:versions=$keep_versions\n" .
-	"  recycle:touch=True\n" .
-	"  recycle:exclude=*.tmp|*.temp|*.o|*.obj|~\$*\n" .
-	"  recycle:exclude_dir=/tmp|/temp|/cache\n";
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesVFS mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesVFS
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesVFS	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/50homesVFS	2007-01-06 14:39:33.000000000 -0700
@@ -0,0 +1,10 @@
+{
+    return "" unless scalar keys %$vfs;
+
+    $OUT = "vfs objects = " . (join " ", keys %$vfs) . "\n";
+    foreach $mod (keys %$vfs) {
+	foreach $opt (keys %{$vfs->{$mod}}) {
+	    $OUT .= "  $mod:$opt=$vfs->{$mod}->{$opt}\n";
+	}
+    }
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/00Setup mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/00Setup
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/00Setup	2006-03-16 00:01:15.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/00Setup	2007-01-06 14:32:46.000000000 -0700
@@ -6,4 +6,6 @@
     $key = $ibay->key;
     $OUT .= "\n[$key]\n";
     $OUT .= "comment = " . $ibay->prop('Name');
+
+    $ibay_vfs = ();
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10recyclebin mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10recyclebin
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10recyclebin	2006-03-16 00:01:15.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10recyclebin	2007-01-06 14:34:45.000000000 -0700
@@ -1,14 +1,11 @@
 {
-    return "# Recycle Bin disabled" unless (($ibay->prop('RecycleBin') || 'disabled') eq 'enabled');
+    $OUT = "";
+    return unless (($ibay->prop('RecycleBin') || 'disabled') eq 'enabled');
 
-    $OUT .= "vfs object = recycle:repository recycle:keeptree ";
-    $OUT .= "recycle:versions recycle:touch recycle:exclude recycle:exclude_dir\n";
-    $OUT .= "  recycle:repository=Recycle Bin\n";
-    $OUT .= "  recycle:keeptree=True\n";
-    $OUT .= "  recycle:versions=";
-    $OUT .= (($ibay->prop('KeepVersions') || 'disabled') eq 'enabled') ?
-		"True\n" : "False\n";
-    $OUT .= "  recycle:touch=True\n";
-    $OUT .= "  recycle:exclude=*.tmp|*.temp|*.o|*.obj|~\$*\n";
-    $OUT .= "  recycle:exclude_dir=/tmp|/temp|/cache";
-}
+    $ibay_vfs->{recycle}->{versions} = ($ibay->prop('KeepVersions') || 'disabled') eq 'enabled' ?  "True" : "False";
+    $ibay_vfs->{recycle}->{repository} = "Recycle Bin";
+    $ibay_vfs->{recycle}->{keeptree} = "True";
+    $ibay_vfs->{recycle}->{touch} = "True";
+    $ibay_vfs->{recycle}->{exclude} = "*.tmp|*.temp|*.o|*.obj|~\$*";
+    $ibay_vfs->{recycle}->{exclude_dir} = "/tmp|/temp|/cache";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10shadowcopy mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10shadowcopy
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10shadowcopy	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/10shadowcopy	2007-01-06 14:34:48.000000000 -0700
@@ -0,0 +1,4 @@
+{
+    $ibay_vfs->{shadow_copy} = () if (($ibay->prop('ShadowCopy') || 'disabled') eq 'enabled');
+    $OUT = "";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/90vfs mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/90vfs
--- e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/90vfs	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.14.0/root/etc/e-smith/templates/etc/smb.conf/ibays/90vfs	2007-01-06 14:34:51.000000000 -0700
@@ -0,0 +1,10 @@
+{
+    return "" unless scalar keys %$ibay_vfs;
+
+    $OUT = "ibay_vfs objects = " . (join " ", keys %$ibay_vfs) . "\n";
+    foreach $mod (keys %$ibay_vfs) {
+	foreach $opt (keys %{$ibay_vfs->{$mod}}) {
+	    $OUT .= "  $mod:$opt=$ibay_vfs->{$mod}->{$opt}\n";
+	}
+    }
+}
