diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir
--- e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir	2002-03-12 15:19:38.000000000 -0700
+++ mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-create-profiledir	2005-07-17 22:45:37.661531152 -0600
@@ -27,18 +27,19 @@
 use esmith::util;
 use esmith::db;
 
-tie my %accounts, 'esmith::config', '/home/e-smith/accounts';
+use esmith::AccountsDB;
+my $adb = esmith::AccountsDB->open_ro();
 
 my $event = $ARGV [0];
 
-my @users = grep { db_get_type(\%accounts, $_) eq 'user' } keys %accounts;
+my @users = map { $_->key } $adb->users;
 
 my @newusers = ($event eq "post-upgrade") ? @users : $ARGV[1] ;
 
 foreach my $user ( @newusers )
 {
     die "$user is not a user account\n"
-        unless ( db_get_type(\%accounts, $user) eq "user");
+        unless ( grep /^$user$/ @users );
 
     my $dir = "/home/e-smith/files/samba/profiles/$user";
    
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-delete-profiledir mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-delete-profiledir
--- e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-delete-profiledir	2003-05-20 15:50:00.000000000 -0600
+++ mezzanine_patched_e-smith-samba-1.13.0/root/etc/e-smith/events/actions/user-delete-profiledir	2005-07-17 22:47:45.957869312 -0600
@@ -28,12 +28,14 @@
 use esmith::db;
 use File::Path;
 
-tie my %accounts, 'esmith::config', '/home/e-smith/accounts';
+use esmith::AccountsDB;
+my $adb = esmith::AccountsDB->open_ro();
 
 my $event = $ARGV [0];
 my $account = $ARGV [1];
 
-unless ( db_get_type(\%accounts, $account) eq "user-deleted" )
+$a = $adb->get($account) || undef;
+unless ( defined $a && $a->prop('type') eq "user-deleted" )
 {
     warn "$account is not a user account\n";
     exit (0);
diff -Nur -x '*.orig' -x '*.rej' e-smith-samba-1.13.0/root/etc/rc.d/init.d/supervise/smb mezzanine_patched_e-smith-samba-1.13.0/root/etc/rc.d/init.d/supervise/smb
--- e-smith-samba-1.13.0/root/etc/rc.d/init.d/supervise/smb	2005-07-17 22:49:08.144092504 -0600
+++ mezzanine_patched_e-smith-samba-1.13.0/root/etc/rc.d/init.d/supervise/smb	2005-07-17 22:35:40.658593440 -0600
@@ -3,7 +3,7 @@
 # Source function library.
 . /etc/rc.d/init.d/functions
 
-STATUS=$(/sbin/e-smith/db /home/e-smith/configuration getprop smb status)
+STATUS=$(/sbin/e-smith/db configuration getprop smb status)
 
 case $1 in
     *start)
